src/rust/engine/options/src/parse.rs

```diff
@@ -66,12 +66,16 @@ peg::parser! {
         // NB: ##method(X) is an undocumented peg feature expression that calls input.method(pos, X)
         // (see https://github.com/kevinmehall/rust-peg/issues/283).
         rule quoted_character(quote_char: &'static str) -> char
-            = !(##parse_string_literal(quote_char)) c:$([_]) { c.chars().next().unwrap() }
+            = !(##parse_string_literal(quote_char) / "\\x") c:$([_]) { c.chars().next().unwrap() }
 
         // Python string literal escape sequences.
         // See https://docs.python.org/3/reference/lexical_analysis.html#escape-sequences.
         // Note that only backslash, single-quote, linefeed, carriage return and horizontal tab
         // are also Rust character escape sequences.
+        //
+        // TODO: Support \uXXXX and \UXXXXXXXX escapes? What about \N{name}?
```

Comment:
I'd rather at support for \u and \U in a followup. Thoughts on the need to support \N? Technically we support it today, but that fact is not documented (it falls out of the fact that we eval string-valued options).

Reply:
> I'd rather at support for \u and \U in a followup.

👍 

> Thoughts on the need to support \N?

Hm, good question! `\N` is quite niche... Maybe we can _try_ to not support it? Parse the `\N{...}` but emit an error.

Assuming we do a deprecation period for the switch-over and encourage people to try out the new parser for a release or two before it we delete the Python one... we can hope that people will flag this change in behaviour.

Reply:
It wouldn't be hard to support \N, thanks to the unicode_names crate, but it would increase the size of the pants binary by 500KB.

Reply:
Hahaha, my past comes back to haunt me! https://github.com/huonw/unicode_names hasn't been updated in a long time (and I imagine it's thus using an out-of-date Unicode standard), but I suppose we could resurrect it!